File: feature-select-language/src/main/java/com/andyanika/translator/feature/select/di/SelectLanguageModule.java
package com.andyanika.translator.feature.select.di;

import androidx.lifecycle.ViewModel;

import com.andyanika.resources.ViewModelKey;
import com.andyanika.translator.common.scopes.FragmentScope;
import com.andyanika.translator.feature.select.SelectLanguageViewModel;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import dagger.multibindings.IntoMap;
import io.reactivex.rxjava3.subjects.PublishSubject;
import io.reactivex.rxjava3.subjects.Subject;

@Module
abstract class SelectLanguageModule {
    @Provides
    @FragmentScope
    static Subject<Integer> provideObserver() {
        return PublishSubject.create();
    }

    @Binds
    @IntoMap
    @FragmentScope
    @ViewModelKey(SelectLanguageViewModel.class)
    abstract ViewModel bindViewModel(SelectLanguageViewModel viewModel);
}

File: feature-select-language/src/main/java/com/andyanika/translator/feature/select/di/koin/SelectLanguageModule.kt
package com.andyanika.translator.feature.select.di.koin

import com.andyanika.translator.feature.select.SelectLanguageFragment
import com.andyanika.translator.feature.select.SelectLanguageListAdapter
import com.andyanika.translator.feature.select.SelectLanguageViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val selectLanguageModule = module {

    scope<SelectLanguageFragment> {
        scoped { params -> SelectLanguageListAdapter(params.get()) }
    }

    viewModel {
        SelectLanguageViewModel(get(), get(), get())
    }
}
